### ISSUE TITLE
Readonly Field for Checkboxes?

### ISSUE THREAD

**Issue by ronalfy** [strangerstudios/pmpro-register-helper#168]
I have a use-case where a user would like a field showing for non-logged-in users, and then when viewing in the admin, this field should be readonly.

I've tried with other fields, and readonly works with most, but not all of them.

Here is my code for reference:

```php
function my_pmprorh_init() {

	// don't break if Register Helper is not loaded
	if ( ! function_exists( 'pmprorh_add_registration_field' ) ) {
		return false;
	}

	// define the fields
	$fields = array();

	$fields[] = new PMProRH_Field(
		'maddress',       // input name, will also be used as meta key
		'text',          // type of field
		array(
			'levels'   => 6,
			'label'    => 'Mac Address',
			'required' => true,
			'profile'  => true,
			'location' => 'checkout_boxes',
		)
	);

	if ( ! is_user_logged_in() && ! is_admin() ) {
		$fields[] = new PMProRH_Field(
			'where1',                       // input name, will also be used as meta key
			'checkbox',                     // type of field
			array(
				'levels'   => 6,
				'label'    => 'I understand that my app username and password (which is different from what I am creating now) will appear on my dashboard within 12 hours (usually much sooner). Existing users login information will remain the same.',
				'required' => true,
				'profile'  => false,
				'location' => 'checkout_boxes',
			)
		);
	} else {
		$fields[] = new PMProRH_Field(
			'where1',                       // input name, will also be used as meta key
			'checkbox',                     // type of field
			array(
				'levels'   => 6,
				'label'    => 'I understand that my app username and password (which is different from what I am creating now) will appear on my dashboard within 12 hours (usually much sooner). Existing users login information will remain the same.',
				'required' => false,
				'profile'  => true,
				'location' => 'checkout_boxes',
				'readonly' => true,
			)
		);
	}
	// add the fields into a new checkout_boxes are of the checkout page
	foreach ( $fields as $field ) {
		pmprorh_add_registration_field(
			'checkout_boxes',               // location on checkout page
			$field                      // PMProRH_Field object
		);
	}

	// that's it. see the PMPro Register Helper readme for more information and examples.
}

add_action( 'init', 'my_pmprorh_init' );
```

The profile field does indeed show up in the admin, but the `readonly` attribute for checkboxes is ignored for some reason.

Thanks!

Ronald Huereca

**Comment by andrewlimaza** [comment]
Does the readonly attribute also affect this on the front-end? Or does it only affect when the field is in the admin area?

**Comment by ronalfy** [comment]
@andrewlimaza the readonly attribute, as far as I can tell, is ignored on both the front-end and back-end for checkboxes.

**Comment by ronalfy** [comment]
Marking this as closed as it isn't possible without some obtrusive JS.

https://stackoverflow.com/questions/155291/can-html-checkboxes-be-set-to-readonly

**Comment by andrewlimaza** [comment]
PR incoming, this will be merged in version 1.6
